aliasmgr/aliasmgr.go

```diff
@@ -219,8 +231,16 @@ func (m *Manager) populateMaps() error {
 func (m *Manager) AddLocalAlias(alias, baseScid lnwire.ShortChannelID,
 	gossip bool) error {
 
+	var returnErr error
+
 	m.Lock()
-	defer m.Unlock()
+	defer func() {
+		m.Unlock()
+		err := m.LinkUpdater(baseScid)
+		if err != nil {
+			returnErr = err
```

Comment:
I know this probably works the way you intended that a `defer` can overwrite the content of the variable that's returned.
But this is super hard to understand and you actually need to know a lot about the internals of Go. Is there another way to achieve this?

Why can't the link updater not be called just before the `return` below? Because the lock is still being held?

Reply:
> I know this probably works the way you intended that a defer can overwrite the content of the variable that's returned.

Actually, it would require a check that the `returnErr` is not nil already, otherwise it would shadow a previously generated error. I guess it's too complicated after all

> Why can't the link updater not be called just before the return below? Because the lock is still being held?

this would just remove the "convenience" of unlocking in the defer. Although now that I'm looking at it, I might just remove the defer func altogether.